homeassistant/helpers/script.py

```diff
@@ -429,7 +429,8 @@ async def _async_step(self, log_exceptions):
                         ex, continue_on_error, self._log_exceptions or log_exceptions
                     )
                 else:
-                    self._variables["last_step_error"] = None
+                    if continue_on_error:
+                        self._variables["last_step_error"] = None
```

Comment:
I think we should call `set_error` on the current trace element (instead of adding a variable).

Something like:

```py
async with trace_action(self._hass, self, self._stop, self._variables) as trace_element:
```

Pass `trace_element` to `_handle_exception()` and let it set the error.

This would make the trace similar to how current errors are handled (and adds a uniform way to display it in the frontend as well).

Reply:
This would not be for the frontend but to be able to do some error handling in the next step.

Reply:
> This would not be for the frontend but to be able to do some error handling in the next step (template condition/choose).

If that is the case, I'm working on a general solution for that (which was requested in forums and Twitter threads).

Allowing to provide an alternative sequence when an automation, script, or single sequence step encounters an error. Allowing, e.g., to send out a notification when something goes wrong.